File: src/main/java/com/saybetter/domain/symbol/dao/repository/SymbolReadRepository.java
package com.saybetter.domain.symbol.dao.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

import com.saybetter.domain.symbol.domain.Symbol;

public interface SymbolReadRepository extends JpaRepository<Symbol, Long> {
	List<Symbol> findByTitleIn(List<String> symbols);

	List<Symbol> findByTitleStartingWith(String name);
}

File: storage/storage-mysql-kt/src/main/kotlin/io/say/better/storage/mysql/domains/symbol/repository/SymbolReadRepository.kt
package io.say.better.storage.mysql.domains.symbol.repository

import io.say.better.storage.mysql.domains.symbol.entity.Symbol
import org.springframework.data.jpa.repository.JpaRepository

interface SymbolReadRepository : JpaRepository<Symbol, Long> {
    fun findByTitleContaining(title: String): List<Symbol>

    fun findByTitleIn(symbols: List<String>): List<Symbol>

    fun findByTitleStartingWith(name: String): List<Symbol>
}
